diff --git b/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/AwsCryptographyMaterialProvidersTypes.dfy a/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/AwsCryptographyMaterialProvidersTypes.dfy
index b23ccbeb..c49f959e 100644
--- b/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/AwsCryptographyMaterialProvidersTypes.dfy
+++ a/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/AwsCryptographyMaterialProvidersTypes.dfy
@@ -149,7 +149,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsKeyring == old(History.CreateAwsKmsKeyring) + [DafnyCallEvent(input, output)]
 
@@ -173,7 +175,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsDiscoveryKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsDiscoveryKeyring == old(History.CreateAwsKmsDiscoveryKeyring) + [DafnyCallEvent(input, output)]
 
@@ -198,7 +202,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMultiKeyring == old(History.CreateAwsKmsMultiKeyring) + [DafnyCallEvent(input, output)]
 
@@ -223,7 +229,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsDiscoveryMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsDiscoveryMultiKeyring == old(History.CreateAwsKmsDiscoveryMultiKeyring) + [DafnyCallEvent(input, output)]
 
@@ -247,7 +255,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsMrkKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkKeyring == old(History.CreateAwsKmsMrkKeyring) + [DafnyCallEvent(input, output)]
 
@@ -272,7 +282,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMrkMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkMultiKeyring == old(History.CreateAwsKmsMrkMultiKeyring) + [DafnyCallEvent(input, output)]
 
@@ -296,7 +308,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsMrkDiscoveryKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkDiscoveryKeyring == old(History.CreateAwsKmsMrkDiscoveryKeyring) + [DafnyCallEvent(input, output)]
 
@@ -321,7 +335,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMrkDiscoveryMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkDiscoveryMultiKeyring == old(History.CreateAwsKmsMrkDiscoveryMultiKeyring) + [DafnyCallEvent(input, output)]
 
@@ -336,21 +352,48 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
            )
         && input.keyStore.ValidState()
         && input.keyStore.Modifies !! {History}
+        && ( input.cache.Some?
+             ==> if input.cache.value.Shared? then
+                 && input.cache.value.Shared.ValidState()
+                 && input.cache.value.Shared.Modifies !! {History}
+               else
+                 true)
       modifies Modifies - {History} ,
                (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
                input.keyStore.Modifies ,
+               (if input.cache.Some? then
+                  if input.cache.value.Shared? then
+                    input.cache.value.Shared.Modifies
+                  else
+                    {}
+                else {}) ,
                History`CreateAwsKmsHierarchicalKeyring
       // Dafny will skip type parameters when generating a default decreases clause.
       decreases Modifies - {History} ,
                 (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
-                input.keyStore.Modifies
+                input.keyStore.Modifies ,
+                (if input.cache.Some? then
+                   if input.cache.value.Shared? then
+                     input.cache.value.Shared.Modifies
+                   else
+                     {}
+                 else {})
       ensures
         && ValidState()
         && ( output.Success? ==>
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) - input.keyStore.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {})
+                          - input.keyStore.Modifies
+                          - (if input.cache.Some? then
+                               if input.cache.value.Shared? then
+                                 input.cache.value.Shared.Modifies
+                               else
+                                 {}
+                             else {}) ) )
       ensures CreateAwsKmsHierarchicalKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsHierarchicalKeyring == old(History.CreateAwsKmsHierarchicalKeyring) + [DafnyCallEvent(input, output)]
 
@@ -375,7 +418,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
       ensures CreateAwsKmsRsaKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsRsaKeyring == old(History.CreateAwsKmsRsaKeyring) + [DafnyCallEvent(input, output)]
 
@@ -399,7 +444,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsEcdhKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsEcdhKeyring == old(History.CreateAwsKmsEcdhKeyring) + [DafnyCallEvent(input, output)]
 
@@ -429,7 +476,10 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.generator.Some? then input.generator.value.Modifies else {}) - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.generator.Some? then input.generator.value.Modifies else {})
+                          - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
       ensures CreateMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateMultiKeyring == old(History.CreateMultiKeyring) + [DafnyCallEvent(input, output)]
 
@@ -449,7 +499,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawAesKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawAesKeyring == old(History.CreateRawAesKeyring) + [DafnyCallEvent(input, output)]
 
@@ -469,7 +520,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawRsaKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawRsaKeyring == old(History.CreateRawRsaKeyring) + [DafnyCallEvent(input, output)]
 
@@ -489,7 +541,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawEcdhKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawEcdhKeyring == old(History.CreateRawEcdhKeyring) + [DafnyCallEvent(input, output)]
 
@@ -513,7 +566,9 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.keyring.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.keyring.Modifies ) )
       ensures CreateDefaultCryptographicMaterialsManagerEnsuresPublicly(input, output)
       ensures History.CreateDefaultCryptographicMaterialsManager == old(History.CreateDefaultCryptographicMaterialsManager) + [DafnyCallEvent(input, output)]
 
@@ -543,7 +598,10 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {}) - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {})
+                          - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
       ensures CreateRequiredEncryptionContextCMMEnsuresPublicly(input, output)
       ensures History.CreateRequiredEncryptionContextCMM == old(History.CreateRequiredEncryptionContextCMM) + [DafnyCallEvent(input, output)]
 
@@ -553,17 +611,35 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
       returns (output: Result<ICryptographicMaterialsCache, Error>)
       requires
         && ValidState()
+        && (if input.cache.Shared? then
+              && input.cache.Shared.ValidState()
+              && input.cache.Shared.Modifies !! {History}
+            else
+              true)
       modifies Modifies - {History} ,
+               (if input.cache.Shared? then
+                  input.cache.Shared.Modifies
+                else
+                  {}) ,
                History`CreateCryptographicMaterialsCache
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases Modifies - {History} ,
+                (if input.cache.Shared? then
+                   input.cache.Shared.Modifies
+                 else
+                   {})
       ensures
         && ValidState()
         && ( output.Success? ==>
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.cache.Shared? then
+                               input.cache.Shared.Modifies
+                             else
+                               {}) ) )
       ensures CreateCryptographicMaterialsCacheEnsuresPublicly(input, output)
       ensures History.CreateCryptographicMaterialsCache == old(History.CreateCryptographicMaterialsCache) + [DafnyCallEvent(input, output)]
 
@@ -583,7 +659,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateDefaultClientSupplierEnsuresPublicly(input, output)
       ensures History.CreateDefaultClientSupplier == old(History.CreateDefaultClientSupplier) + [DafnyCallEvent(input, output)]
 
@@ -769,7 +846,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures GetClientEnsuresPublicly(input, output)
       ensures History.GetClient == old(History.GetClient) + [DafnyCallEvent(input, output)]
     {
@@ -790,7 +868,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures GetClientEnsuresPublicly(input, output)
       ensures unchanged(History)
 
@@ -924,7 +1003,8 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     // Given that you are mutating state,
     // your ValidState function is going to get complicated.
 
-    ghost var Modifies: set<object>
+    ghost const Modifies: set<object>
+    ghost var InternalModifies: set<object>
     // For an unassigned field defined in a trait,
     // Dafny can only assign a value in the constructor.
     // This means that for Dafny to reason about this value,
@@ -952,8 +1032,13 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     //    :: m in Modifies - History)              // is in Modifies and really is not History!
 
     predicate ValidState()
-      reads this`Modifies, Modifies - {History}
-      ensures ValidState() ==> History in Modifies
+      ensures ValidState() ==>
+                && History in Modifies
+
+    predicate InternalValidState()
+      reads this`InternalModifies, InternalModifies
+      ensures InternalValidState() ==> History !in InternalModifies
+
     ghost const History: ICryptographicMaterialsCacheCallHistory
     predicate PutCacheEntryEnsuresPublicly(input: PutCacheEntryInput , output: Result<(), Error>)
     // The public method to be called by library consumers
@@ -970,6 +1055,7 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
       ensures PutCacheEntryEnsuresPublicly(input, output)
       ensures History.PutCacheEntry == old(History.PutCacheEntry) + [DafnyCallEvent(input, output)]
     {
+      assume {:axiom} InternalModifies < Modifies && InternalValidState();
       output := PutCacheEntry' (input);
       History.PutCacheEntry := History.PutCacheEntry + [DafnyCallEvent(input, output)];
     }
@@ -977,12 +1063,12 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     method PutCacheEntry' ( input: PutCacheEntryInput )
       returns (output: Result<(), Error>)
       requires
-        && ValidState()
-      modifies Modifies - {History}
+        && InternalValidState()
+      modifies InternalModifies
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases InternalModifies
       ensures
-        && ValidState()
+        && InternalValidState()
       ensures PutCacheEntryEnsuresPublicly(input, output)
       ensures unchanged(History)
 
@@ -1001,6 +1087,7 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
       ensures UpdateUsageMetadataEnsuresPublicly(input, output)
       ensures History.UpdateUsageMetadata == old(History.UpdateUsageMetadata) + [DafnyCallEvent(input, output)]
     {
+      assume {:axiom} InternalModifies < Modifies && InternalValidState();
       output := UpdateUsageMetadata' (input);
       History.UpdateUsageMetadata := History.UpdateUsageMetadata + [DafnyCallEvent(input, output)];
     }
@@ -1008,12 +1095,12 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     method UpdateUsageMetadata' ( input: UpdateUsageMetadataInput )
       returns (output: Result<(), Error>)
       requires
-        && ValidState()
-      modifies Modifies - {History}
+        && InternalValidState()
+      modifies InternalModifies
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases InternalModifies
       ensures
-        && ValidState()
+        && InternalValidState()
       ensures UpdateUsageMetadataEnsuresPublicly(input, output)
       ensures unchanged(History)
 
@@ -1032,6 +1119,7 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
       ensures GetCacheEntryEnsuresPublicly(input, output)
       ensures History.GetCacheEntry == old(History.GetCacheEntry) + [DafnyCallEvent(input, output)]
     {
+      assume {:axiom} InternalModifies < Modifies && InternalValidState();
       output := GetCacheEntry' (input);
       History.GetCacheEntry := History.GetCacheEntry + [DafnyCallEvent(input, output)];
     }
@@ -1039,12 +1127,14 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     method GetCacheEntry' ( input: GetCacheEntryInput )
       returns (output: Result<GetCacheEntryOutput, Error>)
       requires
-        && ValidState()
-      modifies Modifies - {History}
+        // && ValidState()
+        && InternalValidState()
+      modifies InternalModifies - {History}
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases InternalModifies - {History}
       ensures
-        && ValidState()
+        // && ValidState()
+        && InternalValidState()
       ensures GetCacheEntryEnsuresPublicly(input, output)
       ensures unchanged(History)
 
@@ -1063,6 +1153,7 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
       ensures DeleteCacheEntryEnsuresPublicly(input, output)
       ensures History.DeleteCacheEntry == old(History.DeleteCacheEntry) + [DafnyCallEvent(input, output)]
     {
+      assume {:axiom} InternalModifies < Modifies && InternalValidState();
       output := DeleteCacheEntry' (input);
       History.DeleteCacheEntry := History.DeleteCacheEntry + [DafnyCallEvent(input, output)];
     }
@@ -1070,12 +1161,12 @@ module {:extern "software.amazon.cryptography.materialproviders.internaldafny.ty
     method DeleteCacheEntry' ( input: DeleteCacheEntryInput )
       returns (output: Result<(), Error>)
       requires
-        && ValidState()
-      modifies Modifies - {History}
+        && InternalValidState()
+      modifies InternalModifies
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases InternalModifies
       ensures
-        && ValidState()
+        && InternalValidState()
       ensures DeleteCacheEntryEnsuresPublicly(input, output)
       ensures unchanged(History)
 
@@ -1668,7 +1759,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsKeyring == old(History.CreateAwsKmsKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1697,7 +1790,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsDiscoveryKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsDiscoveryKeyring == old(History.CreateAwsKmsDiscoveryKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1727,7 +1822,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMultiKeyring == old(History.CreateAwsKmsMultiKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1757,7 +1854,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsDiscoveryMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsDiscoveryMultiKeyring == old(History.CreateAwsKmsDiscoveryMultiKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1786,7 +1885,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsMrkKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkKeyring == old(History.CreateAwsKmsMrkKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1816,7 +1917,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMrkMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkMultiKeyring == old(History.CreateAwsKmsMrkMultiKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1845,7 +1948,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsMrkDiscoveryKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkDiscoveryKeyring == old(History.CreateAwsKmsMrkDiscoveryKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1875,7 +1980,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
       ensures CreateAwsKmsMrkDiscoveryMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsMrkDiscoveryMultiKeyring == old(History.CreateAwsKmsMrkDiscoveryMultiKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1895,21 +2002,48 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
            )
         && input.keyStore.ValidState()
         && input.keyStore.Modifies !! {History}
+        && ( input.cache.Some?
+             ==> if input.cache.value.Shared? then
+                 && input.cache.value.Shared.ValidState()
+                 && input.cache.value.Shared.Modifies !! {History}
+               else
+                 true)
       modifies Modifies - {History} ,
                (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
                input.keyStore.Modifies ,
+               (if input.cache.Some? then
+                  if input.cache.value.Shared? then
+                    input.cache.value.Shared.Modifies
+                  else
+                    {}
+                else {}) ,
                History`CreateAwsKmsHierarchicalKeyring
       // Dafny will skip type parameters when generating a default decreases clause.
       decreases Modifies - {History} ,
                 (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
-                input.keyStore.Modifies
+                input.keyStore.Modifies ,
+                (if input.cache.Some? then
+                   if input.cache.value.Shared? then
+                     input.cache.value.Shared.Modifies
+                   else
+                     {}
+                 else {})
       ensures
         && ValidState()
         && ( output.Success? ==>
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) - input.keyStore.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {})
+                          - input.keyStore.Modifies
+                          - (if input.cache.Some? then
+                               if input.cache.value.Shared? then
+                                 input.cache.value.Shared.Modifies
+                               else
+                                 {}
+                             else {}) ) )
       ensures CreateAwsKmsHierarchicalKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsHierarchicalKeyring == old(History.CreateAwsKmsHierarchicalKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1939,7 +2073,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
       ensures CreateAwsKmsRsaKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsRsaKeyring == old(History.CreateAwsKmsRsaKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -1968,7 +2104,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.kmsClient.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.kmsClient.Modifies ) )
       ensures CreateAwsKmsEcdhKeyringEnsuresPublicly(input, output)
       ensures History.CreateAwsKmsEcdhKeyring == old(History.CreateAwsKmsEcdhKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -2003,7 +2141,10 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.generator.Some? then input.generator.value.Modifies else {}) - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.generator.Some? then input.generator.value.Modifies else {})
+                          - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
       ensures CreateMultiKeyringEnsuresPublicly(input, output)
       ensures History.CreateMultiKeyring == old(History.CreateMultiKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -2028,7 +2169,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawAesKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawAesKeyring == old(History.CreateRawAesKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -2053,7 +2195,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawRsaKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawRsaKeyring == old(History.CreateRawRsaKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -2078,7 +2221,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateRawEcdhKeyringEnsuresPublicly(input, output)
       ensures History.CreateRawEcdhKeyring == old(History.CreateRawEcdhKeyring) + [DafnyCallEvent(input, output)]
     {
@@ -2107,7 +2251,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - input.keyring.Modifies ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - input.keyring.Modifies ) )
       ensures CreateDefaultCryptographicMaterialsManagerEnsuresPublicly(input, output)
       ensures History.CreateDefaultCryptographicMaterialsManager == old(History.CreateDefaultCryptographicMaterialsManager) + [DafnyCallEvent(input, output)]
     {
@@ -2142,7 +2288,10 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {}) - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {})
+                          - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
       ensures CreateRequiredEncryptionContextCMMEnsuresPublicly(input, output)
       ensures History.CreateRequiredEncryptionContextCMM == old(History.CreateRequiredEncryptionContextCMM) + [DafnyCallEvent(input, output)]
     {
@@ -2157,17 +2306,35 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
       returns (output: Result<ICryptographicMaterialsCache, Error>)
       requires
         && ValidState()
+        && (if input.cache.Shared? then
+              && input.cache.Shared.ValidState()
+              && input.cache.Shared.Modifies !! {History}
+            else
+              true)
       modifies Modifies - {History} ,
+               (if input.cache.Shared? then
+                  input.cache.Shared.Modifies
+                else
+                  {}) ,
                History`CreateCryptographicMaterialsCache
       // Dafny will skip type parameters when generating a default decreases clause.
-      decreases Modifies - {History}
+      decreases Modifies - {History} ,
+                (if input.cache.Shared? then
+                   input.cache.Shared.Modifies
+                 else
+                   {})
       ensures
         && ValidState()
         && ( output.Success? ==>
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History}
+                          - (if input.cache.Shared? then
+                               input.cache.Shared.Modifies
+                             else
+                               {}) ) )
       ensures CreateCryptographicMaterialsCacheEnsuresPublicly(input, output)
       ensures History.CreateCryptographicMaterialsCache == old(History.CreateCryptographicMaterialsCache) + [DafnyCallEvent(input, output)]
     {
@@ -2192,7 +2359,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersService
                && output.value.ValidState()
                && output.value.Modifies !! {History}
                && fresh(output.value)
-               && fresh ( output.value.Modifies - Modifies - {History} ) )
+               && fresh ( output.value.Modifies
+                          - Modifies - {History} ) )
       ensures CreateDefaultClientSupplierEnsuresPublicly(input, output)
       ensures History.CreateDefaultClientSupplier == old(History.CreateDefaultClientSupplier) + [DafnyCallEvent(input, output)]
     {
@@ -2319,7 +2487,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.kmsClient.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.kmsClient.Modifies ) )
     ensures CreateAwsKmsKeyringEnsuresPublicly(input, output)
 
 
@@ -2342,7 +2512,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.kmsClient.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.kmsClient.Modifies ) )
     ensures CreateAwsKmsDiscoveryKeyringEnsuresPublicly(input, output)
 
 
@@ -2366,7 +2538,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
     ensures CreateAwsKmsMultiKeyringEnsuresPublicly(input, output)
 
 
@@ -2390,7 +2564,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
     ensures CreateAwsKmsDiscoveryMultiKeyringEnsuresPublicly(input, output)
 
 
@@ -2413,7 +2589,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.kmsClient.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.kmsClient.Modifies ) )
     ensures CreateAwsKmsMrkKeyringEnsuresPublicly(input, output)
 
 
@@ -2437,7 +2615,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
     ensures CreateAwsKmsMrkMultiKeyringEnsuresPublicly(input, output)
 
 
@@ -2460,7 +2640,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.kmsClient.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.kmsClient.Modifies ) )
     ensures CreateAwsKmsMrkDiscoveryKeyringEnsuresPublicly(input, output)
 
 
@@ -2484,7 +2666,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.clientSupplier.Some? then input.clientSupplier.value.Modifies else {}) ) )
     ensures CreateAwsKmsMrkDiscoveryMultiKeyringEnsuresPublicly(input, output)
 
 
@@ -2499,19 +2683,45 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
                                              && input.branchKeyIdSupplier.value.ValidState()
          )
       && input.keyStore.ValidState()
+      && ( input.cache.Some?
+           ==> if input.cache.value.Shared? then
+               && input.cache.value.Shared.ValidState()
+             else
+               true)
     modifies ModifiesInternalConfig(config) ,
              (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
-             input.keyStore.Modifies
+             input.keyStore.Modifies ,
+             (if input.cache.Some? then
+                if input.cache.value.Shared? then
+                  input.cache.value.Shared.Modifies
+                else
+                  {}
+              else {})
     // Dafny will skip type parameters when generating a default decreases clause.
     decreases ModifiesInternalConfig(config) ,
               (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) ,
-              input.keyStore.Modifies
+              input.keyStore.Modifies ,
+              (if input.cache.Some? then
+                 if input.cache.value.Shared? then
+                   input.cache.value.Shared.Modifies
+                 else
+                   {}
+               else {})
     ensures
       && ValidInternalConfig?(config)
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {}) - input.keyStore.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.branchKeyIdSupplier.Some? then input.branchKeyIdSupplier.value.Modifies else {})
+                        - input.keyStore.Modifies
+                        - (if input.cache.Some? then
+                             if input.cache.value.Shared? then
+                               input.cache.value.Shared.Modifies
+                             else
+                               {}
+                           else {}) ) )
     ensures CreateAwsKmsHierarchicalKeyringEnsuresPublicly(input, output)
 
 
@@ -2535,7 +2745,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.kmsClient.Some? then input.kmsClient.value.Modifies else {}) ) )
     ensures CreateAwsKmsRsaKeyringEnsuresPublicly(input, output)
 
 
@@ -2558,7 +2770,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.kmsClient.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.kmsClient.Modifies ) )
     ensures CreateAwsKmsEcdhKeyringEnsuresPublicly(input, output)
 
 
@@ -2586,7 +2800,10 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.generator.Some? then input.generator.value.Modifies else {}) - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.generator.Some? then input.generator.value.Modifies else {})
+                        - (set m: object, i | i in input.childKeyrings && m in i.Modifies :: m) ) )
     ensures CreateMultiKeyringEnsuresPublicly(input, output)
 
 
@@ -2606,7 +2823,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config) ) )
     ensures CreateRawAesKeyringEnsuresPublicly(input, output)
 
 
@@ -2626,7 +2844,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config) ) )
     ensures CreateRawRsaKeyringEnsuresPublicly(input, output)
 
 
@@ -2646,7 +2865,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config) ) )
     ensures CreateRawEcdhKeyringEnsuresPublicly(input, output)
 
 
@@ -2669,7 +2889,9 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - input.keyring.Modifies ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - input.keyring.Modifies ) )
     ensures CreateDefaultCryptographicMaterialsManagerEnsuresPublicly(input, output)
 
 
@@ -2697,7 +2919,10 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {}) - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.underlyingCMM.Some? then input.underlyingCMM.value.Modifies else {})
+                        - (if input.keyring.Some? then input.keyring.value.Modifies else {}) ) )
     ensures CreateRequiredEncryptionContextCMMEnsuresPublicly(input, output)
 
 
@@ -2709,15 +2934,32 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
     returns (output: Result<ICryptographicMaterialsCache, Error>)
     requires
       && ValidInternalConfig?(config)
-    modifies ModifiesInternalConfig(config)
+      && (if input.cache.Shared? then
+            && input.cache.Shared.ValidState()
+          else
+            true)
+    modifies ModifiesInternalConfig(config) ,
+             (if input.cache.Shared? then
+                input.cache.Shared.Modifies
+              else
+                {})
     // Dafny will skip type parameters when generating a default decreases clause.
-    decreases ModifiesInternalConfig(config)
+    decreases ModifiesInternalConfig(config) ,
+              (if input.cache.Shared? then
+                 input.cache.Shared.Modifies
+               else
+                 {})
     ensures
       && ValidInternalConfig?(config)
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config)
+                        - (if input.cache.Shared? then
+                             input.cache.Shared.Modifies
+                           else
+                             {}) ) )
     ensures CreateCryptographicMaterialsCacheEnsuresPublicly(input, output)
 
 
@@ -2737,7 +2979,8 @@ abstract module AbstractAwsCryptographyMaterialProvidersOperations {
       && ( output.Success? ==>
              && output.value.ValidState()
              && fresh(output.value)
-             && fresh ( output.value.Modifies - ModifiesInternalConfig(config) ) )
+             && fresh ( output.value.Modifies
+                        - ModifiesInternalConfig(config) ) )
     ensures CreateDefaultClientSupplierEnsuresPublicly(input, output)
 
 
