diff --git b/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/AwsCryptographyKeyStoreTypes.dfy a/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/AwsCryptographyKeyStoreTypes.dfy
index 5764797e..4310b660 100644
--- b/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/AwsCryptographyKeyStoreTypes.dfy
+++ a/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/AwsCryptographyKeyStoreTypes.dfy
@@ -1,9 +1,9 @@
 // Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 // Do not modify this file. This file is machine generated, and any changes to it will be overwritten.
-include "../../../../../../../../Volumes/workplace/aws-cryptographic-material-providers-library/StandardLibrary/src/Index.dfy"
-include "../../../../../../../../Volumes/workplace/aws-cryptographic-material-providers-library/ComAmazonawsDynamodb/src/Index.dfy"
-include "../../../../../../../../Volumes/workplace/aws-cryptographic-material-providers-library/ComAmazonawsKms/src/Index.dfy"
+include "../../../../StandardLibrary/src/Index.dfy"
+include "../../../../ComAmazonawsDynamodb/src/Index.dfy"
+include "../../../../ComAmazonawsKms/src/Index.dfy"
 module {:extern "software.amazon.cryptography.keystore.internaldafny.types" } AwsCryptographyKeyStoreTypes
 {
   import opened Wrappers
@@ -785,7 +785,7 @@ abstract module AbstractAwsCryptographyKeyStoreService
   import opened Types = AwsCryptographyKeyStoreTypes
   import Operations : AbstractAwsCryptographyKeyStoreOperations
   function method DefaultKeyStoreConfig(): KeyStoreConfig
-  method KeyStore(config: KeyStoreConfig := DefaultKeyStoreConfig())
+  method {:vcs_split_on_every_assert} {:rlimit 90500000} KeyStore(config: KeyStoreConfig := DefaultKeyStoreConfig())
     returns (res: Result<KeyStoreClient, Error>)
     requires config.ddbClient.Some? ==>
                config.ddbClient.value.ValidState()
